test/Streamly/Test/FileSystem/Handle.hs

```diff
@@ -0,0 +1,149 @@
+-- |
+-- Module      : Streamly.Test.FileSystem.Handle
+-- Copyright   : (c) 2019 Composewell technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+
+module Streamly.Test.FileSystem.Handle (main) where
+
+import Data.Word (Word8)
+import Test.Hspec as H
+import Test.Hspec.QuickCheck
+import Test.QuickCheck (Property, forAll, Gen, vectorOf, choose)
+import Test.QuickCheck.Monadic (monadicIO, assert, run)
+import System.IO
+    ( Handle
+    , IOMode(..)
+    , SeekMode(..)
+    , hClose
+    , hFlush
+    , hSeek
+    , openFile
+    )
+import System.IO.Temp (writeSystemTempFile)
+import Streamly.FileSystem.Handle as H
+
+import qualified Streamly.Data.Fold as FL
+import qualified Streamly.Internal.Data.Stream.IsStream as S
+import qualified Streamly.Internal.Data.Array.Storable.Foreign as A
+import qualified Streamly.Internal.Unicode.Stream as U
+import Data.Functor.Identity (runIdentity)
+import Foreign.Storable (Storable(..))
+
+allocOverhead :: Int
+allocOverhead = 2 * sizeOf (undefined :: Int)
+
+-- XXX this should be in sync with the defaultChunkSize in Array code, or we
+-- should expose that and use that. For fast testing we could reduce the
+-- defaultChunkSize under CPP conditionals.
+--
```

Comment:
Remove this comment. Also, import allocOverhead/defaultChunksize from the Array module?